libs/common/src/types/portal/activity.ts

```diff
@@ -14,7 +14,7 @@ export type PortalActivityLabel = ValuesOf<typeof PORTAL_ACTIVITY_LABEL>;
  * Ideally, we would run some data migration to rename one or both of these fields.
  */
 export type PortalActivity = {
-  type: PortalActivityType;
+  type?: PortalActivityType;
```

Comment:
This feels inconsistent to sometimes not have a `type` - whilst I agree it doesnt feel necessary for deal cancellation we should probably keep to the current pattern & keep this mandatory

Reply:
I'm on both sides of the fence here:

- If we keep it in, we stick to the existing pattern, but we have additional, useless data being created along with unnecessary extra lines of code. This could be confusing I think we've all had a fair amount of confusion RE activities.
- If we make `type` optional and do not add `type` to all activities moving forward, it's misaligned to other activites, but we have one less confusing thing in the code and database.

Ideally we could remove `type` entirely and run data migration for any existing deals.

Also, It seems rare that new activities will be added to the GEF timeline - in which case I imagine/hope the "deal cancelled" activity would be the most recent example to be found

Thoughts?

@Zainzzkk What do you think?

Reply:
I really don't think we should enforce `type` here. In other comments/PRs, people have been confused about the difference between `type` and `label` - moving forward, we don't need both, the only reason for keeping both would be to keep in aligned with legacy activities.

I think having it optional has more benefit than enforcing it.

We shouldn't have every property in `PortalActivityType` as optional - maybe as well use `AnyObject` in that case.

Reply:
PR is approved so marking this as resolved. Happy to revert if there are any strong objections